src/main/java/net/fabricmc/loom/build/ModCompileRemapper.java

```diff
@@ -57,6 +61,19 @@
 
 @SuppressWarnings("UnstableApiUsage")
 public class ModCompileRemapper {
+	// These are placeholders that are used when the actual group/version is missing (null or empty).
+	// This happens when the dependency is a FileCollectionDependency or from a flatDir repository.
+	private static final String MISSING_GROUP = "unspecified";
+	private static final String MISSING_VERSION = "unspecified";
```

Comment:
Should it use the file hash, what will happen now if you replace the input with the same file? it needs to remap any changes.